pahfit/model.py

```diff
@@ -734,158 +758,169 @@ def tabulate(
             # any other iterable will be accepted and converted to array
             wav = np.asarray(wavelengths) * u.micron
 
-        # shift the "observed wavelength grid" to "physical wavelength grid"
-        wav /= 1 + redshift
-        flux_values = flux_function(wav.value)
+        # apply feature mask, make sub model, and set up functional
+        if feature_mask is not None:
+            features_copy = self.features.copy()
+            features_to_use = features_copy[feature_mask]
+        else:
+            features_to_use = self.features
+
+        # if nothing is in range, return early with zeros
+        if len(features_to_use) == 0:
+            return Spectrum1D(
+                spectral_axis=wav, flux=np.zeros(wav.shape) * u.dimensionless_unscaled
+            )
+
+        alt_model = Model(features_to_use)
```

Comment:
Oh wow we make a whole new `Model` just for a component?

Reply:
It can be any combination of components that needs to be tabulated, not necessarily a single component. (Also, remember that fact that `tabulate` was born out of the format `sub_model`). 

This can be changed once we have our standardized component evaluation routines.